File: php/28_Find_the_Index_of_the_First_Occurrence_in_a_String.php
<?php

class Solution
{

    /**
     * @param String $haystack
     * @param String $needle
     * @return Integer
     */
    function strStr($haystack, $needle)
    {

        //$return  = strpos($haystack, $needle);
        //return $return !== false ? $return : -1;

        $m = strlen($needle);
        $n = strlen($haystack);

        for ($windowStart = 0; $windowStart <= $n - $m; $windowStart++) {
            for ($i = 0; $i < $m; $i++) {
                if ($needle[$i] != $haystack[$windowStart + $i]) {
                    break;
                }
                if ($i == $m - 1) {
                    return $windowStart;
                }
            }
        }

        return -1;
    }

}

File: cpp/28_Find_the_Index_of_the_First_Occurrence_in_a_String.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int needleSize = needle.size();
        int haystackSize = haystack.size();

        for (int start = 0; start <= haystackSize - needleSize; start++) {
            for (int i = 0; i < needleSize; i++) {
                if (needle[i] != haystack[start + i]) {
                    break;
                }
                if (i == needleSize - 1) {
                    return start;
                }
            }
        }

        return -1;
    }
};

